core/src/main/java/com/datastrato/gravitino/storage/kv/KvEntityStore.java

```diff
@@ -332,6 +334,20 @@ private List<byte[]> getSubEntitiesPrefix(NameIdentifier ident, EntityType type)
     return prefixes;
   }
 
+  void deleteUserEntitiesIfNecessary(NameIdentifier ident, EntityType type) throws IOException {
+    if (type != METALAKE) {
+      return;
+    }
+    byte[] encode = entityKeyEncoder.encode(ident, type, true);
+    byte[] prefix = replacePrefixTypeInfo(encode, USER.getShortName());
+    transactionalKvBackend.deleteRange(
+        new KvRange.KvRangeBuilder()
+            .start(prefix)
+            .startInclusive(true)
+            .end(Bytes.increment(Bytes.wrap(prefix)).get())
+            .build());
+  }
```

Comment:
What is the purpose of this method, I cannot get it.

Reply:
If we delete a metalake, we should remove the users under the metalake.